File: 0028_implement_strstr.py
# https://leetcode-cn.com/problems/implement-strstr/

class Solution:
    def strStr(self, haystack:str, needle:str) -> int:
        len_haystack = len(haystack)
        len_needle = len(needle)
        if len_needle == 0:
            return 0
           
        if len_haystack < len_needle:
            return -1
        
        res = -1
        for i in range(len_haystack-len_needle+1):
            if haystack[i:i+len_needle] == needle:
                res = i
                break
        return res

File: Cpp/0028_implement_strstr.cpp
// https://leetcode.cn/problems/implement-strstr/

class Solution {
public:
    int strStr(string haystack, string needle) {
        int n_h = haystack.size(), n_n = needle.size();
        if (n_n == 0) {
            return 0;
        }
        for (int i = 0; i <= n_h - n_n; ++i) {
            if (haystack.substr(i, n_n) == needle) {
                return i;
            }
        }
        return -1;
    }
};